### ISSUE TITLE
[HOLD for payment 2023-03-28] Make sure attachments are fetched from appropriate servers

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#15087]
Coming from [here](https://github.com/Expensify/App/issues/14227#issuecomment-1426112286)

### Problem

Chat attachments are loaded from API not matching the client environment. So you can be on Staging app but the chat attachment is loaded from production

### Why is it important

Environment inconsistency can make it harder to find regressions and can introduce some unexpected ones in one deployment cycle

### Solution

Make sure the production App loads the chat attachments from production api and staging from appropriate api server. cc @kidroca

**Comment by MelvinBot** [comment]
Triggered auto assignment to @zanyrenney (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by mountiny** [comment]
@kidroca Do you want to get assigned to this and work hourly for this issue?

**Comment by kidroca** [comment]
> @kidroca Do you want to get assigned to this and work hourly for this issue?

Yes!

**Comment by zanyrenney** [comment]
Chatted to @mountiny in DM - I can unassign as kidroca has their own contract with expensify for bug fixes!

**Comment by kidroca** [comment]
Found another discrepancy here: 

![image](https://user-images.githubusercontent.com/12156624/219037854-07673632-4f99-413a-ac8a-4fe831fadc08.png)

Right now on DEV the staging toggle defaults to ON

on DEV this toggle does not have any effect - we're always using the PROD server 

The **Use Staging Server** toggle only works on staging, where you can use it to toggle between the prod and the staging server

I think if the toggle does not work to change the underlying api (on DEV) we should not render it at all, or at least render it disabled

**Comment by mountiny** [comment]
> I think if the toggle does not work to change the underlying api (on DEV) we should not render it at all, or at least render it disabled

This is correct and it is expected although I agree it is misleading.

On dev you change contents of .env file to change the API. I think we should hide the toggle on dev personally

**Comment by kidroca** [comment]
PR is ready for review: 
- https://github.com/Expensify/App/pull/15178

**Comment by MelvinBot** [comment]
@kidroca Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by kidroca** [comment]
PR ready for review 
- #15178

But we found that it has overlapping with another PR so we started a discussion there

**Comment by MelvinBot** [comment]
@kidroca Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by MelvinBot** [comment]
@kidroca Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by kidroca** [comment]
The PR was on hold, but is unblocked now: 
- https://github.com/Expensify/App/pull/15178

I have to address some minor items from the latest review

**Comment by MelvinBot** [comment]
@kidroca Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by kidroca** [comment]
PR is ready and reviewed
- https://github.com/Expensify/App/pull/15178

@mountiny I think some minor changes dismissed your approval

**Comment by mountiny** [comment]
@zanyrenney sorry assigning you here back as we will have to pay a C+ for testing and reviewing however we would wait for deploy to ensure there are no regression

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.87-1 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/15178


If no regressions arise, payment will be issued on 2023-03-28. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@s77rt] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@s77rt] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@s77rt] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@zanyrenney] Determine if we should create a regression test for the bug.
- [ ] [@s77rt] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to the [appropriate location](https://github.com/Expensify/App/blob/main/contributingGuides/TestRail_Navigation_C%2B.md) to ensure the same bug will not reach production again.
- [ ] [@zanyrenney] Review the proposed regression test steps and location.
- [ ] [@zanyrenney] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by MelvinBot** [comment]
Looks like something related to `react-navigation` may have been mentioned in this issue discussion.

As a reminder, please make sure that all proposals are not workarounds and that any and all attempt to fix the issue holistically have been made before proceeding with a solution. Proposals to change our `DeprecatedCustomActions.js` files should not be accepted.

Feel free to drop a note in [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q) with any questions.

**Comment by s77rt** [comment]
- The PR that introduced the bug has been identified: I think this was implemented this way from the start
- The offending PR has been commented on: n/a ^
- A discussion in #expensify-bugs has been started: Given this is not a regression, should we open a discussion? @mountiny @kidroca

___

**Regression Test Proposal**
1. Login using https://staging.new.expensify.com/
2. Open any chat and send an attachment
3. Verify that the attachment url starts with https://staging.new.expensify.com/chat-attachments/
4. Login using https://new.expensify.com/
5. Open same chat
6. Verify that the attachment url starts with https://new.expensify.com/chat-attachments/

**Comment by mountiny** [comment]
Looking good, thanks! @zanyrenney would you be able to pay $1000 to @s77rt please?

**Comment by mountiny** [comment]
Bumping this one @zanyrenney

**Comment by mountiny** [comment]
Zany is ooo so asking internally for someone to help complete the payment here.

**Comment by dylanexpensify** [comment]
On it - @s77rt here's job to apply to! https://www.upwork.com/jobs/~014a7129dead94e399

**Comment by s77rt** [comment]
@dylanexpensify Applied.

**Comment by dylanexpensify** [comment]
Paid and contract ended! RT issue [here](https://github.com/Expensify/Expensify/issues/273178)! Thanks for playing y'all!

**Comment by lanitochka17** [comment]
Issue reproducible with Windows10/Chrome 

https://github.com/Expensify/App/assets/78819774/23b7bc67-e9ea-42e8-bfb8-ee490ecf11e1

**Comment by zanyrenney** [comment]
didn't mean to close the issue.

**Comment by s77rt** [comment]
@lanitochka17 @zanyrenney I think this is working as expected, the url is pointing to the staging server. Please note that the url can either be `www.expensify.com` or `staging.exepnsify.com` depending on the environment but never `staging.new.expensify.com` nor `new.expensify.com`.

**Comment by kidroca** [comment]
Thanks @s77rt that's correct 

@lanitochka17 when you are on `staging.new.expensify.com` attachments should be served from `staging.expensify.com` exactly how it happens in the video you've attached here: https://github.com/Expensify/App/issues/15087#issuecomment-1671638268

The actual bug we fixed here was that staging attachments were served from `www.expensify.com` and not from `staging.expensify.com` and vice versa

@zanyrenney I think we can close this one
